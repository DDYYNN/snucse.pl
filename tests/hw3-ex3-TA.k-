(* nodeval *)
let tr := makeTree(1, leaf(2), leaf(3)) in
write (nodeVal (lTree(tr)))
;

(* makeRtree + lTree + isEmpty *)
let tr := makeRtree (1, (leaf (2))) in
if isEmpty (lTree (tr)) then write (1) else write (0)
;

(* makeRtree + rTree + nodeVal *)
let tr1 := makeRtree (1, leaf (2)) in
let tr2 := makeTree (3, tr1, leaf (4)) in
let n1 := nodeVal (rTree (lTree (tr2))) in
let n2 := nodeVal (tr2) in
write (n1 + n2)
;

(* dft *)
let t6 := leaf (7) in
let t5 := makeLtree (6, t6) in
let t2 := makeLtree (5, t5) in
let t4 := leaf (4) in
let t3 := makeRtree (2, leaf (3)) in
let t1 := makeTree (1, t3, t4) in
let t0 := makeTree (0, t1, t2) in
dft (t0)
;

(* bft *)
let t6 := leaf (7) in
let t5 := makeLtree (6, t6) in
let t2 := makeLtree (5, t5) in
let t4 := leaf (4) in
let t3 := makeRtree (2, leaf (3)) in
let t1 := makeTree (1, t3, t4) in
let t0 := makeTree (0, t1, t2) in
bft (t0)
;
()
