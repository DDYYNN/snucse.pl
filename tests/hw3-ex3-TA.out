2
1
5
0
1
2
3
4
5
6
7
0
1
5
2
4
6
3
7
